### PR TITLE
feature/p22-41/Tests-for-duration-and-overlapping-timeslots

### PR DESCRIPTION
**Pull request by milicasamardzija**

### CODE REVIEW

File: Tennis-scheduler/src/test/java/com/tennis/tennisscheduler/validators/DurationValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ExtendWith(MockitoExtension.class)
+class DurationValidatorTest {
+
+    @Mock
+    ConstraintValidatorContext constraintValidatorContext;
+    @Mock
+    ConstraintValidator durationValidator;
```
Comment (mirkovicjul) [inline Tennis-scheduler/src/test/java/com/tennis/tennisscheduler/validators/DurationValidatorTest.java:21]: Why Mock annotation here?

File: Tennis-scheduler/src/test/java/com/tennis/tennisscheduler/validators/OverlappingTimeslotsValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.doReturn;
+
+@ExtendWith(MockitoExtension.class)
+class OverlappingTimeslotsValidatorTest {
+
+    @Mock
+    ConstraintValidatorContext constraintValidatorContext;
+    @Mock
+    ConstraintValidator overlappingTimeslotsValidator;
```
Comment (mirkovicjul) [inline Tennis-scheduler/src/test/java/com/tennis/tennisscheduler/validators/OverlappingTimeslotsValidatorTest.java:27]: Why Mock annotation here?

File: Tennis-scheduler/src/test/java/com/tennis/tennisscheduler/validators/DurationValidatorTest.java (hunk 1 above)
Comment (milicasamardzija) [reply Tennis-scheduler/src/test/java/com/tennis/tennisscheduler/validators/DurationValidatorTest.java:21]: It is not needed..I will change it.

File: Tennis-scheduler/src/test/java/com/tennis/tennisscheduler/validators/DurationValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    @Test
+    void durationTooLong(){
+        TimeslotDto timeslotDto = TimeslotDto.builder()
+                .dateStart(new Date(2022, 12,12,18,30))
+                .dateEnd(new Date(2022, 12,12,22,30))
+                .build();
+
+        boolean valid = durationValidator.isValid(timeslotDto, constraintValidatorContext);
+
+        assertFalse(valid);
```
Comment (nemanjaromanic) [inline Tennis-scheduler/src/test/java/com/tennis/tennisscheduler/validators/DurationValidatorTest.java]: You can inline method call as param. Not needed to extract var.

File: Tennis-scheduler/src/test/java/com/tennis/tennisscheduler/validators/DurationValidatorTest.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    @Test
+    void durationTooShort(){
+        TimeslotDto timeslotDto = TimeslotDto.builder()
+                .dateStart(new Date(2022, 12,12,19,30))
+                .dateEnd(new Date(2022, 12,12,19,45))
+                .build();
+
+        boolean valid = durationValidator.isValid(timeslotDto, constraintValidatorContext);
+
+        assertFalse(valid);
```
Comment (nemanjaromanic) [inline Tennis-scheduler/src/test/java/com/tennis/tennisscheduler/validators/DurationValidatorTest.java]: You can inline method call as param. Not needed to extract var.

File: Tennis-scheduler/src/test/java/com/tennis/tennisscheduler/validators/DurationValidatorTest.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    @Test
+    void durationValid(){
+        TimeslotDto timeslotDto = TimeslotDto.builder()
+                .dateStart(new Date(2022, 12,12,18,00))
+                .dateEnd(new Date(2022, 12,12,19,00))
+                .build();
+
+        boolean valid = durationValidator.isValid(timeslotDto, constraintValidatorContext);
+
+        assertTrue(valid);
```
Comment (nemanjaromanic) [inline Tennis-scheduler/src/test/java/com/tennis/tennisscheduler/validators/DurationValidatorTest.java]: You can inline method call as param. Not needed to extract var.

**Review by nemanjaromanic — APPROVED** [review]

**Review by Dzela — APPROVED** [review]

**Pull request merged**